src/util/temp.cpp

```diff
@@ -32,13 +34,29 @@ Result<TempFile> TempFile::create(std::string pattern)
   // hold this result. This is used below as the actual filename.
   std::vector<char> mutable_pattern(pattern.size() + 1);
   ::strncpy(mutable_pattern.data(), pattern.c_str(), mutable_pattern.size());
-  int fd = ::mkostemp(mutable_pattern.data(), O_CLOEXEC);
+  int fd = fn(mutable_pattern.data());
   if (fd < 0) {
     int err = errno;
```

Comment:
Seems fragile to rely on `fn()` to set errno. How about making `fn` return -errno to make it all in-band?

Reply:
Sounds reasonable to me, I will make that change.

Reply:
Reminder for this change.

Reply:
Done